src/main/kotlin/no/nav/familie/klage/kabal/event/BehandlingEventService.kt

```diff
@@ -24,21 +29,38 @@ class BehandlingEventService(
     private val behandlingRepository: BehandlingRepository,
     private val fagsakRepository: FagsakRepository,
     private val personRepository: FagsakPersonRepository,
+    private val stegService: StegService
 ) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
     fun handleEvent(behandlingEvent: BehandlingEvent) {
-        opprettOppgave(behandlingEvent)
-        ferdigstillKlagebehandling()
-    }
-
-    private fun opprettOppgave(behandlingEvent: BehandlingEvent) {
-
         val behandling = behandlingRepository.findByEksternBehandlingIdAndFagsystem(
             behandlingEvent.kildeReferanse,
             behandlingEvent.kilde
         )
+
+        when (behandlingEvent.type) {
+            BehandlingEventType.KLAGEBEHANDLING_AVSLUTTET -> behandleKlageAvsluttet(behandling, behandlingEvent)
+            else -> behandleAnke(behandling, behandlingEvent)
+        }
+    }
+
+    private fun behandleAnke(behandling: Behandling, behandlingEvent: BehandlingEvent) {
+        opprettOppgave(behandlingEvent, behandling)
+    }
+
+    private fun behandleKlageAvsluttet(behandling: Behandling, behandlingEvent: BehandlingEvent) {
+        when (behandling.status) {
+            BehandlingStatus.FERDIGSTILT -> logger.error("Mottatt event på ferdigstilt behandling $behandlingEvent") // TODO korrigeringer - kan vi få det?
+            else -> {
+                opprettOppgave(behandlingEvent, behandling)
+                ferdigstillKlagebehandling(behandling)
+            }
+        }
+    }
+
+    private fun opprettOppgave(behandlingEvent: BehandlingEvent, behandling: Behandling) {
         val fagsakDomain = fagsakRepository.finnFagsakForBehandlingId(behandling.id)
         val personId = fagsakDomain?.fagsakPersonId
```

Comment:
Er dette egentlig `fagsak` og ikke `fagsakDomain` ?

Reply:
FagsakDomain ble introdusert i ef-sak for å skille på selve fagsak og identer, der FagsakDomain er selve DAO, mens fagsak er en aggregering av data

Reply:
`finnFagsakForBehandlingId` burde alltid returnere et verdi, hvis man spør etter fagsak på en behandling som vi har, så skal fagsaken finnes, hvis ikke er det noe feil :)

Reply:
☝️ enig. Det er en feil.